docs/development/visualize/development-create-visualization.asciidoc

```diff
@@ -0,0 +1,307 @@
+- *options.showIndexSelection*: <bool> show or hide index selection (defaults to true)
+- *options.hierarchicalData*: <bool> use hierarchical data (defaults to false) [BWC]
+- *isExperimental*: <bool> mark visualization as experimental (defaults to false)
+
+
+Each of the factories have some of the custom parameters, which will be described below.
+
+[[development-base-visualization-type]]
+==== Base Visualization Type
+The base visualization type does not make any assumptions about the rendering technology you are going to use and
+works with pure javascript. Its the visualization type we recommend to use.
+
+You need to provide an object with constructor, render function which will be called every time options 
+or data change and destroy function which will be called to cleanup.
+ 
+The render function receives the data object and status object which tells what actually changed.
+Render function needs to return a promise, which should be resolved once the visualization is done rendering.
+
+image::images/visualize-flow.png[Main Flow]
+
+
+["source","js"]
+-----------
+import { VisFactoryProvider } from 'ui/vis/vis_factory';
+
+class MyVisualization {
+   constructor(vis, el) {
+      this.el = el;
+      this.vis = vis;
+   }
+   render(visData, status) {
+      return new Promise(resolve => {
+         ...
+         resolve('done rendering');
+      }),
+   destroy() {
+      console.log('destroying');
+   }
+}
+
+export default function MyNewVisType(Private) {
+  const VisFactory = Private(VisFactoryProvider);
+
+  return VisFactory.createBaseVisualization({
+    name: 'my_new_vis',
+    title: 'My New Vis',
+    icon: 'my_icon',
+    description: 'Cool new chart',
+    visualization: MyVisualization
+  });
+}
+-----------
+
+[[development-angular-visualization-type]]
+==== Angular Visualization Type
+Angular visualization type assumes you are using angular as your rendering technology. Instead of providing the
+controller we need to provide the angular template to render. 
+
+Visualization will receive vis, uiState and visData on the $scope and needs to call $scope.renderComplete() once 
+its done rendering.
+
+["source","js"]
+-----------
+export default function MyNewVisType(Private) {
+  const VisFactory = Private(VisFactoryProvider);
+  
+  return VisFactory.createAngularVisualization({
+    name: 'my_new_vis',
+    title: 'My New Vis',
+    icon: 'my_icon',
+    description: 'Cool new chart',
+    visConfig: {
+       template: '<div ng-controller="MyAngularController"></div>`
+    }
+  });
+}
+-----------
+
+[[development-react-visualization-type]]
+==== React Visualization Type
+React visualization type assumes you are using React as your rendering technology. Instead of passing it an angular
+template you need to pass React component.
+
+Visualization will receive vis, uiState and visData as props as well as the renderComplete which should be called once
+done rendering.
+
+["source","js"]
+-----------
```

Comment:
but this code is not really a react component, it just imports the react component